worlds/jakanddaxter/__init__.py

```diff
@@ -0,0 +1,493 @@
+        self.power_cell_thresholds.append(self.options.lava_tube_cell_count.value)
+        self.power_cell_thresholds.append(100)  # The 100 Power Cell Door.
+
+        # Order the thresholds ascending and set the options values to the new order.
+        try:
+            if self.options.enable_ordered_cell_counts:
+                self.power_cell_thresholds.sort()
+                self.options.fire_canyon_cell_count.value = self.power_cell_thresholds[0]
+                self.options.mountain_pass_cell_count.value = self.power_cell_thresholds[1]
+                self.options.lava_tube_cell_count.value = self.power_cell_thresholds[2]
+        except IndexError:
+            pass  # Skip if not possible.
+
+        # For the fairness of other players in a multiworld game, enforce some friendly limitations on our options,
+        # so we don't cause chaos during seed generation. These friendly limits should **guarantee** a successful gen.
+        # We would have done this earlier, but we needed to sort the power cell thresholds first.
+        ap_settings = settings.get_settings()
+        enforce_friendly_options = ap_settings.jakanddaxter_options.enforce_friendly_options
+        if enforce_friendly_options:
+            if self.multiworld.players > 1:
+                from .Rules import enforce_multiplayer_limits
+                enforce_multiplayer_limits(self)
+            else:
+                from .Rules import enforce_singleplayer_limits
+                enforce_singleplayer_limits(self)
+
+        # Calculate the number of power cells needed for full region access, the number being replaced by traps,
+        # and the number of remaining filler.
+        if self.options.jak_completion_condition == Options.CompletionCondition.option_open_100_cell_door:
+            self.total_prog_cells = 100
+        else:
+            self.total_prog_cells = max(self.power_cell_thresholds[:3])
+        non_prog_cells = self.total_power_cells - self.total_prog_cells
+        self.total_trap_cells = min(self.options.filler_power_cells_replaced_with_traps.value, non_prog_cells)
+        self.options.filler_power_cells_replaced_with_traps.value = self.total_trap_cells
+        self.total_filler_cells = non_prog_cells - self.total_trap_cells
+
+        # Verify that we didn't overload the trade amounts with more orbs than exist in the world.
+        # This is easy to do by accident even in a singleplayer world.
+        self.total_trade_orbs = (9 * self.options.citizen_orb_trade_amount) + (6 * self.options.oracle_orb_trade_amount)
+        from .Rules import verify_orb_trade_amounts
+        verify_orb_trade_amounts(self)
+
+        # Cache the orb bundle size and item name for quicker reference.
+        if self.options.enable_orbsanity == Options.EnableOrbsanity.option_per_level:
+            self.orb_bundle_size = self.options.level_orbsanity_bundle_size.value
+            self.orb_bundle_item_name = orb_item_table[self.orb_bundle_size]
```

Comment:
I just opened a PR that breaks this behavior. No guarantee it gets merged but this is a behavior that I don't think should be relied on anyways.
```suggestion
            self.orb_bundle_item_name = orb_item_table[self.orb_bundle_size.value]
```

Reply:
We got the `.value` on the line just before that, don't we? 

```py
            self.orb_bundle_size = self.options.level_orbsanity_bundle_size.value
            self.orb_bundle_item_name = orb_item_table[self.orb_bundle_size]
```

Reply:
oop i conflated it with the line above it i can't be trusted